### PR TITLE
Fixed typos in CLI reference and standardized formatting

### PR DESCRIPTION
**Pull request by donerancl**
Fixed some typos in the CLI reference and made formatting more consistent.

### CODE REVIEW

**Comment by donerancl** [comment]
I have a question. I added empty lines before a few lists of bullet points in cli docstrings, which is required for them to format properly in the docs, but flake8 seems to have a problem with the whitespace. Is this something we should override? @KnathanM 
image

File: chemprop/cli/train.py [hunk 1]
```diff
@@ -439,24 +439,24 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
     split_args.add_argument(
         "--splits-file",
         type=Path,
-        help="Path to a JSON file containing pre-defined splits for the input data, formatted as a list of dictionaries with keys 'train', 'val', and 'test' and values as lists of indices or strings formatted like '0-2,4'. See documentation for more details.",
+        help="Path to a JSON file containing pre-defined splits for the input data, formatted as a list of dictionaries with keys 'train', 'val', and 'test' and values as lists of indices or strings formatted (e.g., '0-2,4')",
```
Comment (KnathanM) [inline chemprop/cli/train.py:442]:
```suggestion
        help="Path to a JSON file containing pre-defined splits for the input data, formatted as a list of dictionaries with keys 'train', 'val', and 'test' and values as lists of indices or strings with formatted ranges (e.g., '0-2,4')",
```

File: chemprop/cli/hpopt.py [hunk 2]
```diff
@@ -119,10 +119,11 @@ def add_hpopt_args(parser: ArgumentParser) -> ArgumentParser:
     Some options are bundles of parameters or otherwise special parameter operations.
 
     Special keywords:
-        basic - the default set of hyperparameters for search: depth, ffn_num_layers, dropout, message_hidden_dim, and ffn_hidden_dim.
-        learning_rate - search for max_lr, init_lr_ratio, final_lr_ratio, and warmup_epochs. The search for init_lr and final_lr values
+
+        - `basic`: Default set of hyperparameters for search (depth, ffn_num_layers, dropout, message_hidden_dim, and ffn_hidden_dim)
+        - `learning_rate`: Search for max_lr, init_lr_ratio, final_lr_ratio, and warmup_epochs. The search for init_lr and final_lr values
             are defined as fractions of the max_lr value. The search for warmup_epochs is as a fraction of the total epochs used.
```
Comment (KnathanM) [inline chemprop/cli/hpopt.py:125]:
```suggestion
        - `learning_rate`: Search for max_lr, init_lr_ratio, final_lr_ratio, and warmup_epochs. The search for init_lr and final_lr values are defined as fractions of the max_lr value. The search for warmup_epochs is as a fraction of the total epochs used.
```

**Review by KnathanM — COMMENTED** [review]
Thanks for working on this Anna. Two small comments

**Comment by jonwzheng** [comment]
I saw this PR and thought I'd add some additional edits. My last couple of commits adds code formatting for keywords that look unobtrusive (helpful e.g. if you're running `chemprop train --help` ) while rendering nicely  on the readthedocs page.   (However they're going to conflict with the suggestions Nathan provided previously; the second one regarding learning_rate has been resolved.)

File: chemprop/cli/hpopt.py [hunk 3]
```diff
@@ -115,15 +115,10 @@ def add_hpopt_args(parser: ArgumentParser) -> ArgumentParser:
...
-            are defined as fractions of the max_lr value. The search for warmup_epochs is as a fraction of the total epochs used.
-        all - include search for all 13 inidividual keyword options
-
+        help=f"""The model parameters over which to search for an optimal hyperparameter configuration. Some options are bundles of parameters or otherwise special parameter operations. Special keywords include:
+        - ``basic``: Default set of hyperparameters for search (depth, ffn_num_layers, dropout, message_hidden_dim, and ffn_hidden_dim)
+        - ``learning_rate``: Search for max_lr, init_lr_ratio, final_lr_ratio, and warmup_epochs. The search for init_lr and final_lr values are defined as fractions of the max_lr value. The search for warmup_epochs is as a fraction of the total epochs used.
+        - ``all``: Include search for all 13 individual keyword options (including: activation, aggregation, aggregation_norm, and batch_size which aren't included in the other two keywords).
     Individual supported parameters:
         {list(DEFAULT_SEARCH_SPACE.keys())}
```
Comment (jonwzheng) [inline chemprop/cli/hpopt.py:123]: There's an out-standing bug here. If `ray` isn't installed, the keys of `DEFAULT_SEARCH_SPACE` are never populated. Since it's an optional install not required when the docs are built, the "Individual supported parameters" is always empty.

We could just leave this here until `ray` becomes a mandatory install, or ignore it and try to fix it later, or just comment it out for now (it would be nice to have a persistently updated list of search space keys in case it changes later)

Comment (jonwzheng) [reply chemprop/cli/hpopt.py:123]: Actually this issue looks like it'll be addressed by #918, so no need to worry about it here.

File: chemprop/cli/hpopt.py [hunk 4]
```diff
@@ -253,10 +244,10 @@ def update_args_with_config(args: Namespace, config: dict) -> Namespace:
     for key, value in config.items():
         match key:
             case "final_lr_ratio":
-                setattr(args, "final_lr", value * config.get("max_lr", args.max_lr))
+                setattr(args, "final_lr", value * args.max_lr)
 
             case "init_lr_ratio":
-                setattr(args, "init_lr", value * config.get("max_lr", args.max_lr))
+                setattr(args, "init_lr", value * args.max_lr)
```
Comment (KnathanM) [inline chemprop/cli/hpopt.py:293]: Looks like there is a merge error here that set these lines back to a previous version.

File: chemprop/cli/train.py [hunk 5]
```diff
@@ -382,7 +382,7 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
     train_args.add_argument("--max-lr", type=float, default=1e-3, help="Maximum learning rate.")
     train_args.add_argument("--final-lr", type=float, default=1e-4, help="Final learning rate.")
     train_args.add_argument(
-        "--epochs", type=int, default=50, help="the number of epochs to train over"
+        "--epochs", type=int, default=50, help="The number of epochs to train over."
```
Comment (KnathanM) [inline chemprop/cli/train.py:368]: Leaving incomplete sentences as lowercase and not punctuated was our plan for formatting which would undo this change. What do you think of that? Happy to hear your editorial thoughts on our style.

File: chemprop/cli/hpopt.py [hunk 6]
```diff
@@ -239,9 +233,6 @@ def process_hpopt_args(args: Namespace) -> Namespace:
 
 def build_search_space(search_parameters: list[str], train_epochs: int) -> dict:
     if "warmup_epochs" in search_parameters and SEARCH_SPACE.get("warmup_epochs", None) is None:
-        assert (
-            train_epochs >= 6
-        ), "Training epochs must be at least 6 to perform hyperparameter optimization for warmup_epochs."
```
[the remark this answers is not in the document]
Comment (jonwzheng) [reply chemprop/cli/hpopt.py:286]: Thanks for catching this. This must have been from a merge error.

File: chemprop/cli/hpopt.py (hunk 4 above)
Comment (jonwzheng) [reply chemprop/cli/hpopt.py:293]: Thanks, you're right - this is from a merge error.

File: chemprop/cli/common.py [hunk 7]
```diff
@@ -58,24 +58,24 @@ def add_common_args(parser: ArgumentParser) -> ArgumentParser:
...
         default="V2",
         choices=list(AtomFeatureMode.keys()),
-        help="""Choices for multi-hot atom featurization scheme. This will affect both non-reatction and reaction feturization (case insensitive):
+        help="""Choices for multi-hot atom featurization scheme. This will affect both non-reaction and reaction feturization (case insensitive):
 
-- `V1`: Corresponds to the original configuration employed in the Chemprop V1
-- `V2`: Tailored for a broad range of molecules, this configuration encompasses all elements in the first four rows of the periodic table, along with iodine. It is the default in Chemprop V2.
-- `ORGANIC`: This configuration is designed specifically for use with organic molecules for drug research and development and includes a subset of elements most common in organic chemistry, including H, B, C, N, O, F, Si, P, S, Cl, Br, and I.""",
+- ``V1``: Corresponds to the original configuration employed in the Chemprop V1
+- ``V2``: Tailored for a broad range of molecules, this configuration encompasses all elements in the first four rows of the periodic table, along with iodine. Default in Chemprop V2.
```
[the remark this answers is not in the document]
Comment (jonwzheng) [reply chemprop/cli/common.py:77]: I see, that makes sense. Happy to go with the complete sentence form.

File: chemprop/cli/hpopt.py (hunk 4 above)
Comment (jonwzheng) [reply chemprop/cli/hpopt.py:293]: I'll resolve the merge errors.

File: chemprop/cli/train.py (hunk 5 above)
Comment (jonwzheng) [reply chemprop/cli/train.py:368]: We can go ahead and do that, I don't have a strong preference. This change was to just make it consistent with the helpstrings for other functions. The question is - do we leave this in with this PR, or undo it?

**Comment by jonwzheng** [comment]
issues from my previous commits should be fixed now @KnathanM, thanks for catching them

File: chemprop/cli/train.py (hunk 5 above)
Comment (KnathanM) [reply chemprop/cli/train.py:368]: Looking at other changes in this PR, I see there are many that remove the period but keep the capital of incomplete sentences (such as `"Aggregation mode to use during graph predictor"`). Can you remove the period here and in the other instances of incomplete sentences (such as `"Final learning rate."`), but keep the capitals? I also see Anna changed `"the weight to apply ..."` to `"Weight to apply ..."`. To be consistent here should probably be `"Number of epochs ..."`, but I don't have much of an opinion about that. (side note that `"Weight to apply ..."` should probably be `"Weights to apply ..."`, but I have no opinion if that should be changed in this PR or another one.)

Comment (jonwzheng) [reply chemprop/cli/train.py]: Thanks for the suggestion, that's a good catch. I addressed this in the latest commit, let me know what you think. I ended up making those wording changes you suggest in the 2nd half of your comment as well because I noticed some other spots where the consistency could also be improved, so I thought we'd just address them all here.

File: chemprop/cli/train.py [hunk 8]
```diff
@@ -352,31 +352,31 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
         "--task-weights",
         nargs="+",
         type=float,
-        help="the weight to apply to an individual task in the overall loss",
+        help="Weights to apply to an individual task in the overall loss",
```
Comment (KnathanM) [inline chemprop/cli/train.py:355]:
```suggestion
        help="Weights to apply for whole tasks in the loss function",
```
I feel this should reference `tasks` instead of `task` as you need to do something like `--task-weights 1, 1, 5`, i.e. suppling weights for every task. Not sure my wording is the best.

**Comment by jonwzheng** [comment]
@donerancl sorry for hijacking your PR; is there anything you would want to change further / any thoughts on the discussion in this thread?
 
@KnathanM, does the current state of this PR work for you?

**Comment by KnathanM** [comment]
The current state of the PR works for me. There are the merge conflicts that need to be resolved, but I am happy to help with that if needed.

As I reviewed this, I found that the output of `chemprop -h` isn't well formatted. We have a couple help messages that are formatted with new lines, etc. for the rst read the docs, but by default, ArgParse will remove the new lines. This is an issue for a different PR though.

**Comment by jonwzheng** [comment]
Thanks, Nathan! @donerancl is there anything left that we should resolve?

**Review by KnathanM — APPROVED** [review]
